src/apps/opening-hours-editor/EventForm.tsx

```diff
@@ -4,124 +4,246 @@
+          id="event-form-end-time"
+          type="time"
+          value={endTime}
+          onChange={(e) => setEndTime(e.target.value)}
+          min={startTime}
+          max="00:00"
+        />
+        {isRepeatedOpeningHour && (
+          <>
+            <div className="opening-hours-editor-form__checkbox">
+              <input
+                id="event-form-repeated"
+                data-cy="opening-hours-editor-form-repeated"
+                type="checkbox"
+                checked={isRepeated}
+                onChange={(e) => setIsRepeated(e.target.checked)}
+              />
+              <label
+                className="opening-hours-editor-form__label"
+                htmlFor="event-form-repeated"
+              >
+                {t("openingHoursEventFormRepeatedText", {
+                  placeholders: { "@startDate": startDateString }
+                })}
+              </label>
+            </div>
+
+            <div className="opening-hours-editor-form__checkbox">
+              <input
+                id="event-form-weekly"
+                data-cy="opening-hours-editor-form-week-day"
+                type="checkbox"
+                checked={isRepeated}
+                disabled={!isRepeated}
+              />
+              <label className="" htmlFor="event-form-weekly">
+                {t("openingHoursEventFormWeklyText")}
+              </label>
+            </div>
+
+            <div className="opening-hours-editor-form__checkbox">
+              <input
+                id="event-form-week-day"
+                data-cy="opening-hours-editor-form-week-day"
+                type="checkbox"
+                checked={isRepeated}
+                disabled={!isRepeated}
+              />
+              <label className="" htmlFor="event-form-week-day">
+                {weekDayName}
+              </label>
+            </div>
+
+            <label
+              className="opening-hours-editor-form__label"
+              htmlFor="event-form-end-date"
+            >
+              {t("openingHoursEventFormEndDateText")}
+            </label>
+            <input
+              type="date"
+              className="opening-hours-editor-form__time-input"
+              id="event-form-end-date"
+              min={getStringForDateInput(startDate)}
+              disabled={!isRepeated}
+              required={isRepeated}
+              value={repeatedEndDate || ""}
+              onChange={(e) => setRepeatedEndDate(e.target.value)}
+            />
+          </>
+        )}
+        <button
+          data-cy="opening-hours-editor-form-submit"
+          type="submit"
+          className="opening-hours-editor-form__submit"
+          disabled={isSameTime}
+        >
+          {t("openingHoursEventFormSubmitText")}
+        </button>
+        {children}
+      </form>
+
+      <Dialog closeDialog={closeDialog} ref={dialogRef}>
+        {dialogContent}
+      </Dialog>
```

Comment:
Please consider what is going on with this dialog within the form.

As far as I can tell it is not used. `dialogContent` is never assigned.